onnx/checker.cc

```diff
@@ -360,6 +360,81 @@ void check_graph(
   }
 }
 
+void check_function(
+    const FunctionProto& function,
+    const CheckerContext& ctx,
+    const LexicalScopeContext& parent_lex) {
+  enforce_non_empty_field(function, name);
+  enforce_has_field(function, since_version);
+
+  std::unordered_set<std::string> output_names;
```

Comment:
lol it's indeed very very similar to checking graph

Reply:
Yes. they're having similar structure.